ion/parse_text_simple.go

```diff
@@ -56,15 +57,43 @@ func (p *parser) parseBinary(annotations []Symbol) Value {
 func (p *parser) parseLongString(annotations []Symbol) String {
 	text := []byte{}
 	for item := p.peekNonComment(); item.Type == lex.IonStringLong; item = p.peekNonComment() {
-		text = append(text, doStringReplacements(p.next().Val)...)
+		text = append(text, p.doStringReplacements(p.next().Val)...)
 	}
 
 	return String{annotations: annotations, text: text}
 }
 
+// decodeHex takes a slice that is a hex-encoded rune (up to 4 bytes large)
+// and decodes it into a slice containing UTF-8 code units.
+func (p *parser) decodeHex(hex []byte) []byte {
+	// must be a power of two and no larger that uint32
+	hexLen := len(hex)
+	if hexLen <= 0 || hexLen > 8 || hexLen%2 != 0 {
+		// calling code must give us a proper slice...
+		p.panicf("Unexpected hex slice handed to decoder: %v", hex)
+	}
+
+	// decode the hex string into a UTF-32 scalar
+	var cp uint64
+	buf := make([]byte, utf8.UTFMax)
+	for i := 0; i < hexLen; i += 2 {
+		octet, ok := strconv.ParseUint(string(hex[i:i+2]), 16, 8)
+		if ok != nil {
+			// invariant technically violated as the lexer should have validated this
+			p.panicf("bad hex literal not processed by lexer: %v", hex)
```

Comment:
Suggestion: `bad hex literal not caught by the lexer: %v`

Reply:
Fixed.